primitives/src/authorization.rs

```diff
@@ -0,0 +1,41 @@
+use crate::identity_id::IdentityId;
+use codec::{Decode, Encode};
+use rstd::prelude::Vec;
+
+/// Authorization data for two step prcoesses.
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, PartialOrd, Ord)]
+pub enum AuthorizationData {
+    /// Authorization to transfer a ticker
+    TransferTicker(Vec<u8>),
+    /// Any other authorization
+    Custom(Vec<u8>),
+    /// No authorization data
+    None,
```

Comment:
I'm curious as for what was the reason not to use an `Option` type for this.

Reply:
I do not expect to use None ever in the actual world. In fact, I did not even define it in the schema. However, I added it because I needed something to be the default type and we might need none for some unique case.

Using an `Option` will add 1 bit of extra data to all the encoded values and gain us nothing.

Reply:
If `None` is the member of the enum, you can assume that somebody, maybe not you, will eventually use it. I'd say the [Chekhov's gun](https://en.wikipedia.org/wiki/Chekhov%27s_gun) principle is fully applicable here. ;-)

`Option::None` is encoded as the byte `0`, and the `Option::Some` as the byte `1`. This sounds like a reasonable tradeoff which is amortised against the size of the vectors.